addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1178 @@
+        return {
+            'name': _('Opportunity'),
+            'view_type': 'form',
+            'view_mode': 'tree, form',
+            'res_model': 'crm.lead',
+            'domain': [('lead_type', '=', 'opportunity')],
+            'res_id': self.id,
+            'view_id': False,
+            'views': [(form_view.id, 'form'),
+                      (tree_view.id, 'tree'), (False, 'kanban'),
+                      (False, 'calendar'), (False, 'graph')],
+            'type': 'ir.actions.act_window',
+            'context': {'default_lead_type': 'opportunity'}
+        }
+
+    @api.multi
+    def redirect_lead_view(self):
+        # Get lead views
+        self.ensure_one()
+        form_view = self.env.ref('crm.crm_case_form_view_leads')
+        tree_view = self.env.ref('crm.crm_case_tree_view_leads')
+        return {
+            'name': _('Lead'),
+            'view_type': 'form',
+            'view_mode': 'tree, form',
+            'res_model': 'crm.lead',
+            'domain': [('lead_type', '=', 'lead')],
+            'res_id': self.id,
+            'view_id': False,
+            'views': [(form_view.id, 'form'),
+                      (tree_view.id, 'tree'),
+                      (False, 'calendar'), (False, 'graph')],
+            'type': 'ir.actions.act_window',
+        }
+
+    @api.multi
+    def action_schedule_meeting(self):
+        """
+        Open meeting's calendar view to schedule meeting on current opportunity.
+        :return dict: dictionary value for created Meeting view
+        """
+        self.ensure_one()
+        res = self.env['ir.actions.act_window'].for_xml_id('calendar', 'action_calendar_event')
+        partner_ids = [self.env.user.partner_id.id]
+        if self.partner_id:
+            partner_ids.append(self.partner_id.id)
+        res['context'] = {
+            'search_default_opportunity_id': self.lead_type == 'opportunity' and self.id or False,
+            'default_opportunity_id': self.lead_type == 'opportunity' and self.id or False,
+            'default_partner_id': self.partner_id.id,
+            'default_partner_ids': partner_ids,
+            'default_team_id': self.team_id.id,
+            'default_name': self.name,
+        }
+        return res
+
+    @api.multi
+    def action_set_lost(self):
+        """ Lost semantic: probability = 0, active = False """
+        return self.write({'probability': 0, 'active': False})
+    # Backward compatibility
+    case_mark_lost = action_set_lost
+
+    @api.multi
+    def action_set_active(self):
+        return self.write({'active': True})
+
+    @api.multi
+    def action_set_unactive(self):
+        return self.write({'active': False})
+
+    @api.multi
+    def action_set_won(self):
+        """ Won semantic: probability = 100 (active untouched) """
+        stages_leads = {}
+        for lead in self:
+            stage_id = lead.stage_find(lead.team_id.id or False, [('probability', '=', 100.0), ('on_change', '=', True)])
```

Comment:
no need to use False, team_id is recordset(s).
like this way: 
  lead.stage_find(lead.team_id.id,  [('probability', '=', 100.0), ('on_change', '=', True)])